### ISSUE TITLE
Directions returning points out of roads

### ISSUE THREAD

**Issue by natanlaurett** [GIScience/openrouteservice#340]
Hello there. I'm using the Directions API to build paths between two points, and I think I might have found a bug in the database or in the API. When I run the tests with two points in some rural (or suburban) areas, frequently I get points out of roads. That wouldn't be such a worry, but since the Directions API must return only coordinates on roads, I'm reporting the problem. See the following examples:

Belo Horizonte, Minas Gerais Brazil (this is a state capital, heavly populated): I have run a test using a point in the downtown and another one in a neighborhood a little bit far away from there. As you can see, most of the points are indeed on the roads. However, when I get some points like (lat)-19.797748,  (lon) -43.976409, they seem a little bit deslocated.

[https://api.openrouteservice.org/directions?api_key=**YOUR-API-KEY**&coordinates=-43.96458%2C-19.86766%7C-43.9786%2C-19.7951&profile=driving-hgv&preference=fastest&format=json&units=m&language=en&geometry=true&geometry_format=geojson&geometry_simplify=true&instructions=true&instructions_format=text&roundabout_exits=&attributes=&maneuvers=&radiuses=&bearings=&continue_straight=&elevation=&extra_info=&optimized=true&options=%7B%7D&id=](url)

Piúma, Espírito Santo, Brazil (this is a smaller city, mostly rural): in a similar way, I get some points such as (lat) -20.841622 (lon) -40.746737, therefore out of the roads.

[https://api.openrouteservice.org/directions?api_key=**YOUR-API-KEY**=-40.7477623867004%2C-20.8423044519095%7C-40.7139990265329%2C-20.8380792895057&profile=driving-hgv&preference=fastest&format=json&units=m&language=en&geometry=true&geometry_format=geojson&geometry_simplify=true&instructions=true&instructions_format=text&roundabout_exits=&attributes=&maneuvers=&radiuses=&bearings=&continue_straight=&elevation=&extra_info=&optimized=true&options=%7B%7D&id=](url)

I believe it might be a problem in the database, since some of the largest cities I tested (such as São Paulo, Buenos Aires, New York) seem to not have this issue.

**Comment by rabidllama** [comment]
Hi @Natansl - do you have an image showing where these are off as I have just created the routes using the API and QGIS but I cannot see where these dislocations occur:
![image](https://user-images.githubusercontent.com/7126643/47567681-8867e280-d92f-11e8-8b1b-0307546a4132.png)
The right-most point on that image is the one you said is off the road (-20.841622, -40.746737) whereas in this example it lies exactly at an intersection

**Comment by rabidllama** [comment]
I should add that the start and end points are normally a bit "off the route" as the algorithm has to snap these points to the nearest road

**Comment by natanlaurett** [comment]
Hello again. So, after your response, I took the coordinates that I reported and verified them on OpenStreetMaps. Turns out that the API is working perfectly well, and is returning points in the very center of the roads we're working with. I was getting points out of the Streets because I was testing them on a mapping tool called HamsterMap, which seems to have this problem that I've just mentioned here.

I'm sorry for the trouble, and once again, thank you very much for the swift response.
